tests/node_test.cc

```diff
@@ -820,6 +822,133 @@ TEST_CASE("Node is checked for 2D case", "[node][2D]") {
         REQUIRE(node->acceleration(Nphase)(i) ==
                 Approx(acceleration(i)).epsilon(Tolerance));
     }
+
+    SECTION("Check general velocity constraints") {
+      // Check momentum
+      Eigen::Matrix<double, Dim, 1> momentum;
+      for (unsigned i = 0; i < momentum.size(); ++i) momentum(i) = 10.;
+
+      // Check initial momentum
+      for (unsigned i = 0; i < momentum.size(); ++i)
+        REQUIRE(node->momentum(Nphase)(i) == Approx(0.).epsilon(Tolerance));
+
+      // Check assign momentum to 10
+      REQUIRE(node->update_momentum(false, Nphase, momentum) == true);
+      for (unsigned i = 0; i < momentum.size(); ++i)
+        REQUIRE(node->momentum(Nphase)(i) == Approx(10.).epsilon(Tolerance));
+
+      // Check zero velocity
+      for (unsigned i = 0; i < Dim; ++i)
+        REQUIRE(node->velocity(Nphase)(i) == Approx(0.).epsilon(Tolerance));
+
+      // Check mass
+      mass = 100.;
+      node->update_mass(false, Nphase, mass);
+      REQUIRE(node->mass(Nphase) == Approx(100.).epsilon(Tolerance));
+
+      // Compute and check velocity
+      node->compute_velocity();
+      for (unsigned i = 0; i < Dim; ++i)
+        REQUIRE(node->velocity(Nphase)(i) == Approx(0.1).epsilon(Tolerance));
+
+      // Check acceleration
+      Eigen::Matrix<double, Dim, 1> acceleration;
+      for (unsigned i = 0; i < acceleration.size(); ++i) acceleration(i) = 5.;
+      REQUIRE(node->update_acceleration(false, Nphase, acceleration) == true);
+      for (unsigned i = 0; i < acceleration.size(); ++i)
+        REQUIRE(node->acceleration(Nphase)(i) == Approx(5.).epsilon(Tolerance));
+
+      // Check velocity before constraints
+      Eigen::Matrix<double, Dim, 1> velocity;
+      velocity << 0.1, 0.1;
+      for (unsigned i = 0; i < velocity.size(); ++i)
+        REQUIRE(node->velocity(Nphase)(i) ==
+                Approx(velocity(i)).epsilon(Tolerance));
+
+      // Check acceleration before constraints
+      acceleration << 5., 5.;
+      for (unsigned i = 0; i < acceleration.size(); ++i)
+        REQUIRE(node->acceleration(Nphase)(i) ==
+                Approx(acceleration(i)).epsilon(Tolerance));
+
+      SECTION("Check general velocity constraints in 1 direction") {
```

Comment:
Could we merge this and the Cartesian constraints, so that they are in independent sections?